scheduler/pkg/kafka/pipeline/kafkamanager.go

```diff
@@ -267,13 +324,48 @@ func createResponseErrorPayload(modelName string, response []byte) []byte {
 	return append([]byte(modelName+" : "), response...)
 }
 
+func createRebalanceCb(km *KafkaManager, pipeline *Pipeline) kafka.RebalanceCb {
+	logger := km.logger.WithField("func", "createRebalanceCb")
+	return func(consumer *kafka.Consumer, ev kafka.Event) error {
+		switch e := ev.(type) {
+		case kafka.AssignedPartitions:
+			km.mu.Lock()
+			km.mu.Unlock()
+
+			logger.Debug("Rebalance: Assigned partitions:", e.Partitions)
+			err := consumer.Assign(e.Partitions)
+			if err != nil {
+				pipeline.consumer.partitions = nil
+				return fmt.Errorf("assign error: %w", err)
+			}
+
+			// Update the pipeline consumer partitions
+			pipeline.consumer.partitions = make([]int32, len(e.Partitions))
+			for i, partition := range e.Partitions {
+				pipeline.consumer.partitions[i] = partition.Partition
+			}
+		case kafka.RevokedPartitions:
+			km.mu.Lock()
+			km.mu.Unlock()
+
+			logger.Debug("Rebalance: Revoked partitions:", e.Partitions)
+			err := consumer.Unassign()
```

Comment:
Here, aren't we assuming that all partitions have been unassigned? (rather than just e.Partitions?)

Reply:
We are dealing only with partitions per consumers ... or maybe I don't understand the question ...